### ISSUE TITLE
Disable polling with force update leaves gitrepo in not ready status in 2.9

### ISSUE THREAD

**Issue by mmartin24** [rancher/fleet#3048]
### Steps to reproduce.

- Deploy latest rancher 2.9-head
- On local push a gitrepo (which later will be modified) with flag `disablePollling: true`. I use this one: https://github.com/fleetqa/fleet-qa-examples-public/blob/main/disable-polling/nginx.yaml setting replicas to 2.
- Once repo is deployed change something on the repo. (In the example above, change replicas to 5)
- Click `Force Update`

### Observed behavior:
- Replicas are correctly changed to 5 in deployments but gitrepo remains in `Not Ready` state and clusters are not ready either (0/1).

Log message on fleet-local description is:
```
 Message:           NotReady(1) [Cluster fleet-local/local]; deployment.apps disable-polling-nginx-ns/nginx-test-polling [progressing] Deployment does not have minimum availability., Available: 2/5 
```

https://github.com/user-attachments/assets/85f53424-5970-4086-ae1e-95497247b052

### Expected behavior:
- Replicas should be correctly modified and gitrepo should be correctly changed to Active and clusters set to 1/1

Tested on `v2.9-023781e52103c45a9bbdf3cb47d60b9336321daa-head` and `fleet:104.1.1+up0.10.5-rc.2` but happens with a bit earlier versions

**Comment by manno** [comment]
If this can only be triggered by being extremely fast, let's document a workaround instead of a complicated fix.

**Comment by kkaempf** [comment]
Does this also happen in 2.10 ?

**Comment by mmartin24** [comment]
> Does this also happen in 2.10 ?

No. This was a regression on 2.9 only, in. In 2.10 it is ok (example [here](https://github.com/rancher/fleet-e2e/actions/runs/12346018858/job/34457311298#step:9:395))

**Comment by kkaempf** [comment]
fixed in 2.10, closing.
